Abandoned layers no longer regenerate. Layers stay in order after abandoning and drilling more. Target new layer shows depth. Layer ore density is configurable.

XenoTheStrange opened the pull request:
Fixed behavior of abandoned layers being re-generated the next time the player digs down. Ensured the list remains in order.
Added depth reading to "Target: New Layer" in UI so that it will read "Target: New layer at depth: ##m"

Now players can keep digging down without having all of the previous layers loaded.

XenoTheStrange commented:
Previous behavior example
- Dig down to layer at 30m
- Abandon layer at 20m
- Drill new layer
It would re-create the layer at 20m

New behavior: drill continues to the next layer that hasn't been drilled, ignoring the layer you abandoned and not magically regenerating it.

XenoTheStrange committed:
Fixed regenerating layers + ordering, added depth to "dig new layer" UI

Fixed behavior of abandoned layers being re-generated the next time the player digs down.
Ensured the list remains in order.
Added depth reading to "Target: New Layer" in UI

XenoTheStrange committed:
Added Configurable Ore Density

Added entry to mod configuration menu that allows players to control how much minable ore (on average) will be in newly mined layers.

XenoTheStrange commented:
Added option to configure new layer ore density.

XenoTheStrange committed:
Translations for previous changes

DeepL has a pretty easy-to-use api

XenoTheStrange commented:
Used deepL api for translations. Am curious how Mlie does it (because you have to change xml keys without translating them. I just pasted the stuff in manually but I can imagine a script which compares xml files for differences and re-translates stuff when there's a change.

emipa606 commented:
You cannot change the existing keys for translation. That will make them not work for the previous versions of the game (<1.4) 
If you need to change keys you need to create new keys and use the new translation there.

XenoTheStrange commented:
> You cannot change the existing keys for translation. That will make them not work for the previous versions of the game (<1.4) If you need to change keys you need to create new keys and use the new translation there.

O I didn't know that. Ok.

XenoTheStrange committed:
Revert "Translations for previous changes"

This reverts commit 7d0bdcb2f28493398e1eaf5001e7d1b2cd5471a5.

emipa606 commented:
Since I wont be able to change the code in the previous assemblies they will still use the same keys making their translations look wierd if changed

XenoTheStrange committed:
Fixed translations / OreDensity

I didn't think about how old dlls were using the same translation files. Added new entry to avoid conflict.

Made sure to re-add ore-density entries...

XenoTheStrange commented:
I added new entries so there won't be conflicts anymore 👌

XenoTheStrange committed:
Set default OreDensity

Because I forgot that the first time

XenoTheStrange committed:
Changed deepest layer detection, fixed ui not updating

Reworked how I implemented the check for the next layer etc.
Can now be added to a save which had the previous version and it'll fix the internal values.

XenoTheStrange commented:
- Ensured UI new depth variable will update to (lowest layer currently dug + 1) when loaded into an existing save that used an older version of the mod.

XenoTheStrange commented on Source/DeepRim/GenStep_RocksFromGrid_Deep.cs:
```diff
@@ -57,7 +57,7 @@ public override void Generate(Map map, GenStepParams parms)
 
         var genStep_ScatterLumpsMineable = new GenStep_ScatterLumpsMineable();
-        var num = 16f;
+        var num = DeepRimMod.instance.DeepRimSettings?.OreDensity != null ? (float)DeepRimMod.instance.DeepRimSettings?.OreDensity : 16f;
```
I don't know if the line needs to be this long but I was getting hard exceptions when generating a new layer if I didn't check for null 🤷.
Is because OreDensity doesn't have a getter so it might actually be null if the user hasn't gone into that menu.... idk.

16f is the default so that's what it'll be if it would have crashed.

XenoTheStrange committed:
Edit log messages

XenoTheStrange committed:
(VSCode Config)

This is for me so I don't have to trouble myself with it again. Can be ignored.

XenoTheStrange committed:
changed files so vscode will build for me

self-explanatory. can be ignored.

XenoTheStrange committed:
Added map size 125

The gap between 75 and 200 tiles is just too big. Added 125.

At some point this should be changed so the player can select any number and maybe the size indicator (Large, small, incident) could update when that value falls in a certain range.

XenoTheStrange committed:
Fix Mine Shafts Ignoring Adjacent Storage

Mine shafts would not detect adjacent storage which shared a space with a conduit or other building.
Now it sees all storages around the mine shaft.

XenoTheStrange committed:
Fix underground maps remaining when the colony is abandoned

Added a check in Tick() for underground lifts that will destroy the layer's map if the surfaceMap stops existing.

XenoTheStrange commented:
Hopefully those last two are the only problems left for me to run into xD.
Last one also fixed if a player put down a mine inside of a camp and left it the layers would have remained.

XenoTheStrange commented:
I'll reopen this in a bit when I'm done messing around with it. I'm renaming branches. Should have done it earlier.
